src/main/java/reactor/netty/transport/ProxyProvider.java

```diff
@@ -324,6 +317,57 @@ public ProxyProvider build() {
 		}
 	}
 
+	static final class RegexShouldProxyPredicate implements Predicate<SocketAddress> {
+
+		private final String regex;
+		private Pattern pattern;
+
+        /**
+         * Create a {@link RegexShouldProxyPredicate} based off the provided regular expression.
+         * @param regex The string regular expression
+         * @return a predicate whether we should direct to proxy
+         */
+		public static RegexShouldProxyPredicate fromRegularExpression(String regex) {
+            return new RegexShouldProxyPredicate(regex);
+        }
+
+        /**
+         * Creates a {@link RegexShouldProxyPredicate} based off the provided pattern with possible wildcards as
+         * described in https://docs.oracle.com/javase/7/docs/api/java/net/doc-files/net-properties.html
+         *
+         * @param pattern The string wildcarded expression
+         * @return a predicate whether we should direct to proxy
+         */
+        public static RegexShouldProxyPredicate fromWildcardedPattern(String pattern) {
+            return new RegexShouldProxyPredicate(pattern.replaceAll("\\.", "\\\\.").replaceAll("\\*", "\\.*"));
```

Comment:
Here https://docs.oracle.com/javase/8/docs/technotes/guides/net/proxies.html there is a little bit more specific about `http.nonProxyHosts` 

```
http.nonProxyHosts:a list of hosts that should be reached directly, bypassing the proxy. This is a list of patterns separated by '|'. The patterns may start or end with a '*' for wildcards. Any host matching one of these patterns will be reached through a direct connection instead of through a proxy.
```

So `*` is accepted only as a start/end of the pattern.

Let's add tests like `localhost|127.*|[::1]`, `127.*`, `127.*.0.1`, `some.foo.*`, `some.*.com`

Reply:
true this was a bit of a hack to get the other boilerplate through, I'll follow up with something more legitimate 😄

Reply:
Are there problems with using ```DefaultProxySelector``` from ```sun.net.spi```? It handles transforming the string to java pattern in ```disjunctToRegex```

Reply:
`sun.net.spi` I don't think we want to use the internal java packages